components/txn_types/src/lock.rs

```diff
@@ -95,9 +98,30 @@ impl Lock {
             b.push(MIN_COMMIT_TS_PREFIX);
             b.encode_u64(self.min_commit_ts.into_inner()).unwrap();
         }
+        if self.use_async_commit {
+            b.push(ASYNC_COMMIT_PREFIX);
+            b.encode_var_u64(self.secondaries.len() as _).unwrap();
+            for k in &self.secondaries {
+                b.encode_compact_bytes(k).unwrap();
+            }
+        }
         b
     }
 
+    fn pre_allocate_size(&self) -> usize {
+        let mut size =
```

Comment:
It's unsound. Some fields are missing, like `for_update_ts`, `txn_size`, `min_commit_ts`.

Reply:
It's just the same code as the old code. Maybe it was always incorrect. I'll check it again.

Reply:
Its not unsound, its just sub-optimal because the Vec will be reallocated